### PR TITLE
Fix part of #5070: Display empty answer message in ratio input interaction

### PR DESCRIPTION
**Pull request by theMr17**
## Explanation

Fixes part of #5070

Enables the `submit_answer_button` when the pending answer is empty. Instead of disabling the button, an error message, stating "_**Enter a ratio to continue.**_", is now displayed when the user attempts to submit a blank answer.

`RatioInputInteractionViewTestActivity` is added to accessibility-label exemption as this is a test activity.

https://github.com/oppia/oppia-android/assets/84731134/33c797c5-af44-4eaa-980e-fafaccf72d26

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by adhiamboperes** [comment]
@theMr17 is this ready for a full pass or are you looking for a preliminary review on approach? Please convert to ready and assign to me if the former.

**Comment by theMr17** [comment]
@adhiamboperes, please do a preliminary review of the approach. I will proceed to write tests after that.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/RatioExpressionInputInteractionViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/RatioExpressionInputInteractionViewModel.kt]: Maybe add a comment explaining why this is `true`.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/RatioExpressionInputInteractionViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/RatioExpressionInputInteractionViewModel.kt]: can you add here the argument names so it is clear what is null/true, so that your code is self-documenting.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml]:
```suggestion
  <string name="ratio_error_empty_input">Enter a ratio to continue.</string>
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
@theMr17, I have left some suggestions, but the overall approach looks good to me.

**Comment by theMr17** [comment]
@adhiamboperes PTAL

**Review by adhiamboperes — APPROVED** [review]
Thanks @theMr17! This LGTM. 

Thanks for moving the ratio tests to their own suite as well.

**Comment by adhiamboperes** [comment]
@BenHenning, this PR is blocked on codeowner, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt]: Please directly import assertThat (for consistency with other tests and per Truth convention).

File: app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt]: Espresso functions like onView & others should also be directly imported. Ditto for ViewMatchers, ViewAssertions, and ViewActions.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @theMr17! Really sorry about the late review here. I just had two small comments, otherwise the PR LGTM.

**Comment by theMr17** [comment]
@BenHenning PTAL

File: app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt]: This has been resolved in the latest changes.

File: app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/testing/RatioInputInteractionViewTestActivityTest.kt]: This has been resolved in the latest changes.

**Review by BenHenning — APPROVED** [review]
Thanks @theMr17! Please make sure to respond to addressed comments on future PRs. Otherwise the PR LGTM.

**Comment by theMr17** [comment]
> Thanks @theMr17! Please make sure to respond to addressed comments on future PRs. Otherwise the PR LGTM.

Thanks @BenHenning, I will surely keep that in mind.

**Pull request merged**
